src/firefly/java/edu/caltech/ipac/firefly/util/ImageSetConverter.java

```diff
@@ -0,0 +1,209 @@
+package edu.caltech.ipac.firefly.util;
+
+import edu.caltech.ipac.firefly.server.util.DsvToDataGroup;
+import edu.caltech.ipac.firefly.server.util.Logger;
+import edu.caltech.ipac.util.DataGroup;
+import edu.caltech.ipac.util.DataObject;
+import edu.caltech.ipac.util.DataType;
+import edu.caltech.ipac.util.download.FailedRequestException;
+import org.apache.commons.csv.CSVFormat;
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.HashMap;
+import java.util.List;
+
+
+/**
+ * Created by zhang on 9/27/17.
+ * This class will read in the ImageSet information table (maybe from an API) and
+ * the convert to a JSONObject.
+ */
+public class ImageSetConverter {
```

Comment:
A couple of things about this class.
- should not be responsible for converting into JSONObject.  should only create  List<ImageMasterDataEntry>
- why is it in a util package?  nothing generic about this class.  it's an extension of IrsaMasterDataSource.  should be inside IrsaMasterDataSource.java or in the same package.  if it's in a separate file, it should have 'irsa' in its name as well.

Reply:
So, you are saying that this should be the 'Combiner' and another class should convert the combined entries into json, is that what i hear? I agree.

Reply:
Emmanuel suggested to make it as a utility class.  Where should I put now?  Please give suggestions!  Thanks!

Reply:
I don't understand this class either. JSON is the transport layer that is handled laster by ImageMasterData.  This point of this class should it create a `List<ImageMasterDataEntry>`.  There should be no json involved.

Reply:
The ticket was misleading because it had 2 ideas in one but you did part of it by using json directly in the server-side. Between Loi and I, i think we've explained the differences to Lijun and the (new) approach to keep the java side to handle the setters of the values from the master table (IRSA specific btw). The idea behind the ticket was to have a parser to generate objects that would end into json in the client. The only part needed is the parser into java objects that can be sent to client on server (command) request. 
Does it make more sense now? @lznakano